src/renderers/dom/shared/eventPlugins/ChangeEventPlugin.js

```diff
@@ -260,34 +119,32 @@ function handleControlledInputBlur(inst, node) {
 var ChangeEventPlugin = {
   eventTypes: eventTypes,
 
-  _isInputEventSupported: isInputEventSupported,
-
-  extractEvents: function(
-    topLevelType,
-    targetInst,
-    nativeEvent,
-    nativeEventTarget,
-  ) {
+  extractEvents(topLevelType, targetInst, nativeEvent, nativeEventTarget) {
     var targetNode = targetInst
       ? ReactDOMComponentTree.getNodeFromInstance(targetInst)
       : window;
 
-    var getTargetInstFunc, handleEventFunc;
-    if (shouldUseChangeEvent(targetNode)) {
-      getTargetInstFunc = getTargetInstForChangeEvent;
-    } else if (isTextInputElement(targetNode)) {
-      if (isInputEventSupported) {
-        getTargetInstFunc = getTargetInstForInputOrChangeEvent;
-      } else {
-        getTargetInstFunc = getTargetInstForInputEventPolyfill;
-        handleEventFunc = handleEventsForInputEventPolyfill;
+
+    // On the selectionchange event, the target is the document which isn't
+    // helpful becasue we need the input, so we use the activeElement instead.
+    if (!isTextInputEventSupported && topLevelType === 'topSelectionChange') {
```

Comment:
it does generally, The null check is out of caution, since this depends on on the input being focused, there may be edge cases where focus drops or some programatic driving of the browser doesn't set focus, etc. I don't think there is much we can do about such edge cases and wanted to limit the possible effect to only cases where the polyfill was being used.